core/src/main/java/io/kestra/core/tasks/flows/WaitFor.java

```diff
@@ -0,0 +1,238 @@
+package io.kestra.core.tasks.flows;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.executions.NextTaskRun;
+import io.kestra.core.models.executions.TaskRun;
+import io.kestra.core.models.flows.State;
+import io.kestra.core.models.hierarchies.AbstractGraph;
+import io.kestra.core.models.hierarchies.GraphCluster;
+import io.kestra.core.models.hierarchies.RelationType;
+import io.kestra.core.models.tasks.FlowableTask;
+import io.kestra.core.models.tasks.ResolvedTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.FlowableUtils;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.GraphUtils;
+import io.kestra.core.utils.TruthUtils;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.slf4j.Logger;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Run a specific task until the expected result.",
+    description = """
+        Used to wait an HTTP response or a job to end.
+        You can use the variable `previous` to access the previous task output.
+        The variable `previous` will be null at first iteration.
+        Conditions is always check after the task execution.
+        Output of the child task will be accessible through the outpout of the WaitFor task.
+        """
+)
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            title = "Wait for a task to return a specific output",
+            code = """
+                id: waitFor
+                namespace: myteam
+                                
+                tasks:
+                  - id: waitfor
+                    type: io.kestra.core.tasks.flows.WaitFor
+                    task:
+                      id: return
+                      type: io.kestra.core.tasks.debugs.Return
+                      format: "{{ outputs.waitfor.iterationCount }}"
+                    condition: "{{ outputs.return.value != '4' }}"
```

Comment:
```suggestion
                id: exampleFlow
                namespace: myteam
                                
                tasks:
                  - id: waitFor
                    type: io.kestra.core.tasks.flows.WaitFor
                    condition: "{{ outputs.return.value != '4' }}"
                    task:
                      id: return
                      type: io.kestra.core.tasks.debugs.Return
                      format: "{{ outputs.waitFor.iterationCount }}"
```
just looks visually better to show condition at the top

Reply:
I agree that what I did at first (probably also in flow for test)
But I thought it would be more clear for ppl to have the condition after the task (we execute the task then we test the condition) but its up to you!